src/zenml/cli/utils.py

```diff
@@ -800,11 +807,222 @@ def convert_structured_str_to_dict(string: str) -> Dict[str, str]:
     )
 
 
+def _parse_json_if_applicable(key: str, raw: str) -> Any:
+    """Parse JSON strings starting with { or [, otherwise return raw value.
+
+    Args:
+        key: The key/flag name (used for error messages).
+        raw: The raw string value to parse.
+
+    Returns:
+        Parsed JSON object if the string is valid JSON starting with { or [,
+        otherwise returns the original string.
+
+    Raises:
+        click.ClickException: If the string appears to be JSON but is malformed.
+    """
+    if isinstance(raw, str) and raw and raw[0] in "[{":
+        # Security check: limit JSON size
+        if len(raw) > MAX_JSON_ARGUMENT_SIZE:
+            raise click.ClickException(
+                f"JSON for --{key} exceeds maximum size of {MAX_JSON_ARGUMENT_SIZE} bytes"
+            )
+
+        try:
+            # Parse JSON with standard decoder
+            parsed = json.loads(raw)
+
+            # Check depth after parsing
+            def check_depth(obj: Any, current_depth: int = 0) -> None:
+                if current_depth > MAX_JSON_DEPTH:
+                    raise json.JSONDecodeError(
+                        f"Maximum JSON depth ({MAX_JSON_DEPTH}) exceeded",
+                        raw,
+                        0,
+                    )
+                if isinstance(obj, dict):
+                    for value in obj.values():
+                        check_depth(value, current_depth + 1)
+                elif isinstance(obj, list):
+                    for item in obj:
+                        check_depth(item, current_depth + 1)
+
+            check_depth(parsed)
+            return parsed
+        except json.JSONDecodeError as e:
+            # Check if this is a potentially sensitive field to avoid exposing secrets
+            sensitive_field_patterns = {
+                "api_key",
```

Comment:
What you're trying to do here is maintain a list of a secret fields of all stack components, but that is a bad idea. Not only will no one even know that this exists, and developers that create or modify stack components will not include their fields here. Additionally, this also includes some stack-component specific knowledge in some general-purpose CLI function which is not a good idea.